RxJavaTemplate/domain/src/main/java/co/nimblehq/rxjava/domain/storage/NormalSharedPreferences.kt

```diff
@@ -0,0 +1,24 @@
+package co.nimblehq.rxjava.domain.storage
+
+import android.content.Context
+import android.content.SharedPreferences
+import javax.inject.Inject
+
+class NormalSharedPreferences @Inject constructor(
+    private val applicationContext: Context,
+    private val defaultSharedPreferences: SharedPreferences
+) : BaseSharedPreferences() {
+
+    init {
+        useDefaultSharedPreferences()
+    }
+
+    fun useDefaultSharedPreferences() {
+        sharedPreferences = defaultSharedPreferences
+    }
+
+    // Use this function for creating a custom sharedPreferences if needed
+    fun useCustomSharedPreferences(name: String) {
+        sharedPreferences = applicationContext.getSharedPreferences(name, Context.MODE_PRIVATE)
+    }
```

Comment:
Could we make these functions private? 🙏

Reply:
@manh-t this is intentional since in the future, we might need to use it for specific usecases. Please ignore the warnings here.  🙏